modules/base/rendering/grids/renderableradialgrid.cpp

```diff
@@ -0,0 +1,461 @@
+        _minRadius.setMaxValue(_maxRadius);
+    });
+
+    _minRadius.onChange([&]() {
+        _gridIsDirty = true;
+        _maxRadius.setMinValue(_minRadius);
+    });
+
+    addProperty(_maxRadius);
+    addProperty(_minRadius);
+}
+
+bool RenderableRadialGrid::isReady() const {
+    return _gridProgram != nullptr;
+}
+
+void RenderableRadialGrid::initializeGL() {
+    _gridProgram = BaseModule::ProgramObjectManager.request(
+        ProgramName,
+        []() -> std::unique_ptr<ghoul::opengl::ProgramObject> {
+            return global::renderEngine.buildRenderProgram(
+                ProgramName,
+                absPath("${MODULE_BASE}/shaders/grid_vs.glsl"),
+                absPath("${MODULE_BASE}/shaders/grid_fs.glsl")
+            );
+        }
+    );
+}
+
+void RenderableRadialGrid::deinitializeGL() {
+    BaseModule::ProgramObjectManager.release(
+        ProgramName,
+        [](ghoul::opengl::ProgramObject* p) {
+            global::renderEngine.removeRenderProgram(p);
+        }
+    );
+    _gridProgram = nullptr;
+}
+
+void RenderableRadialGrid::render(const RenderData& data, RendererTasks&){
+    _gridProgram->activate();
+
+    _gridProgram->setUniform("opacity", _opacity);
+
+    const glm::dmat4 modelTransform =
+        glm::translate(glm::dmat4(1.0), data.modelTransform.translation) * // Translation
+        glm::dmat4(data.modelTransform.rotation) *  // Spice rotation
+        glm::scale(glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale));
+
+    const glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() *
+                                          modelTransform;
+
+    _gridProgram->setUniform("modelViewTransform", modelViewTransform);
+    _gridProgram->setUniform(
+        "MVPTransform",
+        glm::dmat4(data.camera.projectionMatrix()) * modelViewTransform
+    );
+    
+    _gridProgram->setUniform("gridColor", _gridColor);
+
+    float adjustedLineWidth = 1.f;
+
+#ifndef __APPLE__
+    adjustedLineWidth = _lineWidth;
+#endif
+
+    // Saves current state:
+    GLboolean isBlendEnabled = glIsEnabledi(GL_BLEND, 0);
+    GLfloat currentLineWidth;
+    glGetFloatv(GL_LINE_WIDTH, &currentLineWidth);
+    GLboolean isLineSmoothEnabled = glIsEnabled(GL_LINE_SMOOTH);
+
+    GLenum blendEquationRGB;
+    GLenum blendEquationAlpha;
+    GLenum blendDestAlpha;
+    GLenum blendDestRGB;
+    GLenum blendSrcAlpha;
+    GLenum blendSrcRGB;
+    glGetIntegerv(GL_BLEND_EQUATION_RGB, &blendEquationRGB);
+    glGetIntegerv(GL_BLEND_EQUATION_ALPHA, &blendEquationAlpha);
+    glGetIntegerv(GL_BLEND_DST_ALPHA, &blendDestAlpha);
+    glGetIntegerv(GL_BLEND_DST_RGB, &blendDestRGB);
+    glGetIntegerv(GL_BLEND_SRC_ALPHA, &blendSrcAlpha);
+    glGetIntegerv(GL_BLEND_SRC_RGB, &blendSrcRGB);
+
+    // Changes GL state:
+    glLineWidth(adjustedLineWidth);
+    glEnablei(GL_BLEND, 0);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glEnable(GL_LINE_SMOOTH);
+    
+    for (GeometryData &c : _circles) {
```

Comment:
The `&` should attach to the `GeometryData` as it is an attribute of the class (making it a `GeometryData` reference, rather than the variable name